src/timeline/TimelineModel.h

```diff
@@ -189,6 +189,7 @@ class TimelineModel : public QAbstractListModel
     Q_PROPERTY(QStringList widgetLinks READ widgetLinks NOTIFY widgetLinksChanged)
     Q_PROPERTY(int roomMemberCount READ roomMemberCount NOTIFY roomMemberCountChanged)
     Q_PROPERTY(bool isEncrypted READ isEncrypted NOTIFY encryptionChanged)
+    Q_PROPERTY(QString fullyReadEventId READ fullyReadEventId NOTIFY fullyReadEventIdChanged)
```

Comment:
How about instead of storing this in a member variable, you calculate it in the READ function? Then you can get rid of all the visibility checks and signals.

Reply:
This would break functionality when the message you left on is an old-ish one.

Each message is going to be calculating the event id in the READ function, but it's not going to always return the same value. The first messages are going to return the intended id of what was the last message the last time the room was opened. But when the room gets marked as read shortly after being opened, all messages rendered after that, including the message that we last left on, are going to return the id of the *currently* last received message (i.e. current index 0), and not get marked with the line.

Reply:
Hm, I guess we'll see how expensive this is. Otherwise there are solutions for that :3